binaries/daemon/src/spawn.rs

```diff
@@ -149,21 +162,82 @@ pub async fn spawn_node(
                 }
             }
 
-            command.spawn().wrap_err(format!(
-                "failed to run runtime {}/{}",
-                runtime_config.node.dataflow_id, runtime_config.node.node_id
-            ))?
+            command
+                .stdin(Stdio::null())
+                .stdout(Stdio::piped())
+                .stderr(Stdio::piped())
+                .spawn()
+                .wrap_err(format!(
+                    "failed to run runtime {}/{}",
+                    runtime_config.node.dataflow_id, runtime_config.node.node_id
+                ))?
         }
     };
 
+    let log_dir = temp_dir();
+
+    let (tx, mut rx) = mpsc::channel(10);
+    let mut file = File::create(
+        &log_dir.join(PathBuf::from(format!("{dataflow_id}-{node_id}.txt")).with_extension("txt")),
+    )
+    .await
+    .expect("Failed to create log file");
+    let mut stdout_lines =
+        (tokio::io::BufReader::new(child.stdout.take().expect("failed to take stdout"))).lines();
+
+    let stdout_tx = tx.clone();
+
+    // Stdout listener stream
+    tokio::spawn(async move {
+        while let Ok(Some(line)) = stdout_lines.next_line().await {
+            let sent = stdout_tx.send(Some(line)).await;
+            if sent.is_err() {
+                break;
```

Comment:
Not sure if breaking here is a good idea. Doesn't this lead to problems that the pipe becomes full and blocks the stdout writes of the spawned programs?